x/delayedack/eibc.go

```diff
@@ -99,47 +118,64 @@ func (im IBCMiddleware) createDemandOrderFromIBCPacket(fungibleTokenPacketData t
 	}
 	// Verify the original recipient is not a blocked sender otherwise could potentially use eibc to bypass it
 	if im.keeper.BlockedAddr(fungibleTokenPacketData.Receiver) {
-		return nil, fmt.Errorf("%s is not allowed to receive funds", fungibleTokenPacketData.Receiver)
+		return nil, fmt.Errorf("not allowed to receive funds: receiver: %s", fungibleTokenPacketData.Receiver)
 	}
 	// Get the fee from the memo
 	fee := eibcMetaData.Fee
 	// Calculate the demand order price and validate it
 	amountInt, ok := sdk.NewIntFromString(fungibleTokenPacketData.Amount)
 	if !ok || !amountInt.IsPositive() {
-		return nil, fmt.Errorf("Failed to convert amount to positive integer, %s", fungibleTokenPacketData.Amount)
+		return nil, fmt.Errorf("convert amount to positive integer: %s", fungibleTokenPacketData.Amount)
 	}
 	feeInt, ok := sdk.NewIntFromString(fee)
 	if !ok || !feeInt.IsPositive() {
-		return nil, fmt.Errorf("Failed to convert fee to positive integer, %s", fee)
+		return nil, fmt.Errorf("convert fee to positive integer: %s", fee)
 	}
 	if amountInt.LT(feeInt) {
-		return nil, fmt.Errorf("Fee cannot be larger than amount")
+		return nil, fmt.Errorf("fee cannot be larger than amount")
 	}
+
+	/*
+		   In case of timeout/errack:
+		       fee = fee_multiplier*transfer_amount
+		       price = transfer_amount-fee
+		       order is created with (price,fee)
+		       the order creator immediately receives price on fulfillment from fulfiller
+		       when the ack/timeout finalizes, the fulfiller receives the transfer_amount
+
+		       therefore:
+		           fulfiller balance += (transfer_amount - (transfer_amount-fee))
+					   equivalent to += fee_multiplier*transfer_amount
+		           demander balance += (transfer_amount - fee)
+		              equivalent to += (1-fee_multiplier)*transfer_amount
+	*/
 	demandOrderPrice := amountInt.Sub(feeInt).String()
-	// Get the denom for the demand order. If it's a timeout packet
-	// than its simply the denom we tried to send. If it's a receive packet
-	// than it's the IBC denom we've got.
+
 	var demandOrderDenom string
 	var demandOrderRecipient string
-	if rollappPacket.Type == commontypes.RollappPacket_ON_TIMEOUT {
-		demandOrderDenom = fungibleTokenPacketData.Denom
-		demandOrderRecipient = fungibleTokenPacketData.Sender
-	} else if rollappPacket.Type == commontypes.RollappPacket_ON_RECV {
+	// Get the denom for the demand order
+	switch rollappPacket.Type {
+	case commontypes.RollappPacket_ON_TIMEOUT:
+		fallthrough
+	case commontypes.RollappPacket_ON_ACK:
+		demandOrderDenom = fungibleTokenPacketData.Denom      // It's what we tried to send
+		demandOrderRecipient = fungibleTokenPacketData.Sender // and to who we tried to send it to
```

Comment:
I believe the comment here is wrong. it's not who we tried to send it to, but the address which sent it from

Reply:
ah yep good spot